Modify S4828: Clarify recommendations (#1269)
SonarSource/rspec: rules/S4828/python/rule.adoc

include::../recommended.adoc[]

== Sensitive Code Example

[source,python]
----
import os

@app.route("/kill-pid/<pid>")
def send_signal(pid):
    os.kill(pid, 9)  # Sensitive

@app.route("/kill-pgid/<pgid>")
def send_signal(pgid):
    os.killpg(pgid, sig)  # Sensitive
----

== Compliant Solution

[source,python]
----
import os

@app.route("/kill-pid/<pid>")
def send_signal(pid):

>     // Validate the untrusted PID,
>     // With a dynamic approved list for example

Author: ```suggestion
    # Validate the untrusted PID,
    # With a pre-approved list or authorization checks
```

>     validate_pid(pid)

Reviewer: Not sure of how it's usually done, so take it with a pinch of salt: since there is validation, I would expect to see a condition (`if validate_pid(pid): ...`) or something about how an exception will be raised if the PID is not valid.

Author: Good suggestion! it would look more explicit, I agree. I really didn't have a clue on how to give a concise and efficient code sample here.

    os.kill(pid, 9)  # Sensitive

@app.route("/kill-pgid/<pgid>")
def send_signal(pgid):

>     // Validate the untrusted PGID,
>     // With a dynamic approved list for example

Author: ```suggestion
    # Validate the untrusted PGID,
    # With a pre-approved list or authorization checks
```

include::../recommended.adoc[]

== Sensitive Code Example

[source,python]
----
import os

@app.route("/kill-pid/<pid>")
def send_signal(pid):
    os.kill(pid, 9)  # Sensitive

@app.route("/kill-pgid/<pgid>")
def send_signal(pgid):

>     os.killpg(pgid, sig)  # Sensitive

Reviewer: @loris-s-sonarsource What's `sig` here? I guess we could rather put 9 or a constant.

Reviewer: 👍
